Content.Client/Gameplay/GameplayStateBase.cs

```diff
@@ -144,8 +145,34 @@ public IEnumerable<EntityUid> GetClickableEntities(MapCoordinates coordinates, I
             var clickQuery = _entityManager.GetEntityQuery<ClickableComponent>();
             var clickables = _entityManager.System<ClickableSystem>();
 
+            // RMC14: We get a local player
+            var localEntity = _playerManager.LocalEntity;
+            var transformQuery = _entityManager.GetEntityQuery<TransformComponent>();
+            var spriteSystem = _entitySystemManager.GetEntitySystem<SpriteSystem>();
+            var spriteQuery = _entityManager.GetEntityQuery<SpriteComponent>();
+            var transformSystem = _entitySystemManager.GetEntitySystem<SharedTransformSystem>();
+
             foreach (var entity in entities)
             {
+                // RMC14: Filter by InteractionTransparencyComponent (if not ignored)
+                if (!ignoreInteractionTransparency && _entityManager.HasComponent<InteractionTransparencyComponent>(entity.Uid) && localEntity != null)
+                {
+                    if (transformQuery.TryGetComponent(entity.Uid, out var entXform) &&
+                        transformQuery.TryGetComponent(localEntity.Value, out var playerXform) &&
+                        spriteQuery.TryGetComponent(entity.Uid, out var sprite))
+                    {
+                        // RMC14: Get the world position and rotation of the sprite
+                        var (spritePos, spriteRot) = transformSystem.GetWorldPositionRotation(entXform);
+                        // RMC14: Get the sprite's BoundingBox
+                        var spriteBox = spriteSystem.CalculateBounds((entity.Uid, sprite), spritePos, spriteRot, eye.Rotation);
+                        // RMC14: Check if the player's position is inside the BoundingBox
+                        if (spriteBox.Contains(transformSystem.GetMapCoordinates(playerXform).Position))
+                        {
+                            // RMC14: Skipping this object for this client
+                            continue;
+                        }
+                    }
+                }
```

Comment:
You should pull out all of this code and move it to an rmc-specific system, you can make it return a boolean for whether or not to skip (continue) here